model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/InfinispanAuthenticationSessionProviderFactory.java

```diff
@@ -78,25 +92,66 @@ public static int getAuthSessionsLimit(Config.Scope config) {
         return limit <= 0 ? DEFAULT_AUTH_SESSIONS_LIMIT : limit;
     }
 
-
     @Override
     public void postInit(KeycloakSessionFactory factory) {
+        this.remoteCacheInvoker = new RemoteCacheInvoker();
+        keyGenerator = new InfinispanKeyGenerator();
         factory.register(new ProviderEventListener() {
 
             @Override
             public void onEvent(ProviderEvent event) {
                 if (event instanceof PostMigrationEvent) {
-
                     KeycloakModelUtils.runJobInTransaction(factory, (KeycloakSession session) -> {
-
                         registerClusterListeners(session);
-
+                        loadAuthSessionsFromRemoteCaches(session);
                     });
                 }
             }
         });
     }
 
+    private void loadAuthSessionsFromRemoteCaches(KeycloakSession session) {
+        for (String cacheName : remoteCacheInvoker.getRemoteCacheNames()) {
```

Comment:
@graziang For this to work, we need to call `addRemoteCache` on `remoteCacheInvoker`, otherwise there are no caches in it.